### PR TITLE
[TestOnly] `tests` folder reorg

### PR DESCRIPTION
**Pull request by catenocrypt**
## Description

Reorganize test files in the `/tests` folder, into subfolders as follows:
- `chains`: folder for blockchain-specific tests (C++ and C iface). Contains only subfolders.
- `common`: blockchain-agnostic common classes (such as PrivateKey, HDWallet, ...).  C++ iface.
- `interface`: tests for C interface methods

The  `tests` folder contains only very limited number of files (`main.cpp` and `CMakeFiles.txt`).

All moved files show up as modified, meaningful review can be done commit-by-commit, I kept plain moves and other fixes in separate commits (<- recommended practice with many moves).
I post below the script with `git mv` commands for reference/reproducibility.

The `/src` folder should be reorganized in similar manner.

While this PR is pending, merge conflicts can occur if test files are changed in other PRs.

Small part of #2483 .

## How to test

All builds & tests run fine.

## Types of changes

-- Refactor

## Checklist

- [x] Create pull request as draft initially, unless its complete.
- [x] Add tests to cover changes as needed.
- [x] Update documentation as needed.
- [x] If there is a related Issue, mention it in the description.

If you're adding a new blockchain

- [ ] I have read the [guidelines](https://developer.trustwallet.com/wallet-core/newblockchain#integration-criteria) for adding a new blockchain.

### CODE REVIEW

**Comment by catenocrypt** [comment]
Script with the `git mv` commands used to move the files, I post it here for reference/reproducibility.
[mv.sh.txt](https://github.com/trustwallet/wallet-core/files/9733972/mv.sh.txt)

**Comment by catenocrypt** [comment]
CI runs:
- iOS tests fail due to some timeout/crash ?
- Linux tests fail with decreased coverage, 95% -> 94.6%, some tests are not executed ?

File: tests/common/Keystore/StoredKeyTests.cpp [hunk 1]
```diff
@@ -272,27 +272,27 @@ TEST(StoredKey, WalletInvalid) {
...
+    const auto key = StoredKey::load(TESTS_ROOT + "/common/Keystore/Data/livepeer.json");
     EXPECT_EQ(key.type, StoredKeyType::privateKey);
     EXPECT_EQ(key.id, "70ea3601-ee21-4e94-a7e4-66255a987d22");
     EXPECT_EQ(key.accounts[0].coin, TWCoinTypeEthereum);
     EXPECT_EQ(hex(key.payload.decrypt(TW::data("Radchenko"))), "09b4379d9a41a71d94ee36357bccb4d77b45e7fd9307e2c0f673dd54c0558c73");
 }
 
 TEST(StoredKey, LoadPBKDF2Key) {
-    const auto key = StoredKey::load(TESTS_ROOT + "/Keystore/Data/pbkdf2.json");
+    const auto key = StoredKey::load(TESTS_ROOT + "/common/Keystore/Data/pbkdf2.json");
```
Comment (Milerius) [inline tests/common/Keystore/StoredKeyTests.cpp:295]: How about having those common path in the same std::filesystem::path global variable in order to simplify future refactoring? https://en.cppreference.com/w/cpp/filesystem/path

**Review by Milerius — APPROVED** [review]
Good job, minor comment for future refactoring, it's better to have path in the same global variable.

```cpp
namespace fs = std::filesystem;
fs::path gCommonDataFolder = TESTS_ROOT / "common/Keystore/data";

/// load(gCommonDataFolder.c_str() / json_name);
```

**Comment by catenocrypt** [comment]
> ```c++
> namespace fs = std::filesystem;
> fs::path gCommonDataFolder = TESTS_ROOT / "common/Keystore/data";
> ```

1. Having a common place for the test data path is good idea
2. I don't see much advantage using `path` here, `load()` takes string anyways, plus there are no big path operations
3. A global variable doesn't work, as `TEST_ROOT` gets initialized in runtime, it is still empty when globals are initialized
4. Instead, I have created a method for returning test data path

```
const std::string testDataPath(const char* subpath) {
    return TESTS_ROOT + "/common/Keystore/Data/" + subpath;
}
```

**Review by Milerius — APPROVED** [review]

**Review by hewigovens — APPROVED** [review]

**Pull request merged**